Simplenote/NoticeController.swift

```diff
@@ -0,0 +1,118 @@
+import UIKit
+
+class NoticeController {
+    // MARK: Proerties
+    //
+    static let shared = NoticeController()
+
+    private var notices: [Notice] = []
+    private var current: Notice?
+    private let noticePresenter = NoticePresenter()
+
+    private var activeViewIsBeingTouched: Bool = false
+    private var timer: Timer?
+
+    private var isPresenting: Bool {
+        current != nil
+    }
+
+    // MARK: Life Cycle
+    //
+    private init() { }
+
+    func setupNoticeController() {
+        noticePresenter.startListeningToKeyboardNotifications()
+    }
+
+    // MARK: Presenting
+    //
+    func present(_ notice: Notice, withTimer timer: Timer? = nil) {
```

Comment:
What you probably want to do here is to have a `TimerFactory` or `TimerProvider` object injected to the controller (in the init method). This will let you replace it with a mocked version in tests.

`TimerFactory` is an object that would create and return Timers :-) 

The interface can look like this:
```
class TimerFactory {
    func scheduledTimer(with timeInterval: TimeInterval, completion: () -> Void) -> Timer
}
```

Reply:
Makes sense!